File: python/read_file.py
# read_file.py
# This program reads the contents of a file named "input.txt" and prints them to the console.
# Github: https://www.github.com/0x4248/examples
# Licence: GNU General Public License v3.0
# By: 0x4248

# Open the file "input.txt" in read mode and store the contents in a variable called contents
with open("input.txt", "r") as file:
    # Read the contents of the file
    contents = file.read()
    # Print the contents of the file
    print(contents)

File: c/read_file.c
/**
 * @file read_file.c
 * @brief Program to read data from a file
 * @details This program reads the contents of a file named "input.txt" and
 * prints them to the console.
 * @github: https://www.github.com/lewisevans2007/examples
 * @license: GNU General Public License v3.0
 * @author Lewis Evans
 */

#include <stdio.h>

int main() {
    FILE *fptr;
    char ch;

    fptr = fopen("input.txt", "r");
    if (fptr == NULL) {
        printf("Error opening file!");
        return 1;
    }

    while ((ch = fgetc(fptr)) != EOF) {
        printf("%c", ch);
    }

    fclose(fptr);

    return 0;
}
